src/main/java/de/tum/in/www1/artemis/service/notifications/SingleUserNotificationService.java

```diff
@@ -65,6 +72,29 @@ private void notifyRecipientWithNotificationType(Object notificationSubject, Not
         saveAndSend(resultingGroupNotification, notificationSubject);
     }
 
+    /**
+     * Notify all users with available assessments about the finished assessment for an exercise submission.
+     * This is an auxiliary method that finds all relevant users and initiates the process for sending SingleUserNotifications and emails
+     *
+     * @param exercise which assessmentDueDate is the trigger for the notification process
+     */
+    public void notifyUsersAboutAssessedExerciseSubmission(Exercise exercise) {
+        // This process can not be replaces via a GroupNotification (can only notify ALL students of the course)
+        // because we want to notify only the students that have a valid assessed submission.
+
+        Set<StudentParticipation> filteredStudentParticipations = Set
+                .copyOf(studentParticipationRepository.findByExerciseIdWithEagerLegalSubmissionsAndLatestResult(exercise.getId()));
+
+        // Load and assign all studentParticipations with results (this information is needed for the emails later)
+        exercise.setStudentParticipations(filteredStudentParticipations);
+
+        // Find all users that should be notified, i.e. users with an assessed participation
```

Comment:
this comment seems to be wrong, the method below does not filter for assessed participations, please make sure the latest result is available and has a completion date (if this was part of the database query above, mention this in the code comments)

Reply:
I applied the requested changes